### PR TITLE
Add External Service settings to the configuration page

### PR DESCRIPTION
**Pull request by riyenas0925**
## Description
* add External System related code
	* CD_EXTERNAL_SERVICE_SERVICE_USED_FLAG (914)
	* CD_EXTERNAL_SERVICE_SETTING (930)
	* CD_DTL_GITHUB_TOKEN (100)
* encrypt the token value and store it in the db
* add External Service Settings tab to Configuration page
* set the token value to not be visible on the Configuration page

### preview
sep-26-2021_12-42-06

## Type of change
Please insert 'x' one of the type of change.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Refactoring, Maintenance
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/main/java/oss/fosslight/service/impl/SystemConfigurationServiceImpl.java [hunk 1]
```diff
@@ -138,6 +145,34 @@
...
+			tokenAuthList.stream().map(c -> {
+				switch (c.getCdDtlNo()) {
+					case CoConstDef.CD_DTL_GITHUB_TOKEN:
+						String _token = (String) tokenDetailMap.get(CoConstDef.CD_DTL_GITHUB_TOKEN);
+						if(!StringUtil.isEmpty(_token)) {
+							String _encToken = null;
+							try {
+								_encToken = CryptUtil.encryptAES256(_token, CoConstDef.ENCRYPT_DEFAULT_SALT_KEY);
+							} catch (Exception e) {
+								e.printStackTrace();
```
Comment (riyenas0925) [inline src/main/java/oss/fosslight/service/impl/SystemConfigurationServiceImpl.java:162]: log로 변경

File: src/main/webapp/WEB-INF/views/admin/system/configuration-js.jsp [hunk 2]
```diff
@@ -79,6 +79,22 @@
...
+				params["externalServiceFlag"] = "Y";
+
+				var externalServiceDetail = {};
+			<c:forEach var="code" items="${ct:getCodeValues(ct:getConstDef('CD_EXTERNAL_SERVICE_SETTING'))}" varStatus="status">
+				externalServiceDetail["${code[0]}"] = $("#external${code[0]}").val();
+			</c:forEach>
+				params["externalServiceDetail"] = externalServiceDetail;
+
+			} else {
+				params["externalServiceFlag"] = "N"
```
Comment (riyenas0925) [inline src/main/webapp/WEB-INF/views/admin/system/configuration-js.jsp:96]: 세미콜론 추가

File: db/initdb.d/fosslight_create.sql [hunk 3]
```diff
@@ -2277,6 +2278,7 @@ INSERT INTO `T2_CODE_DTL` (`CD_NO`, `CD_DTL_NO`, `CD_DTL_NM`, `CD_SUB_NO`, `CD_D
 	('909', '911', 'SMTP Flag', '', 'N', 2, 'Y'),
 	('909', '912', 'Menu Auth Flag', '', 'Y', 3, 'Y'),
 	('909', '913', 'file Path Flag', '', 'Y', 4, 'Y'),
+	('909', '914', 'External Service Flag', '', 'N', 5, 'Y'),
```
Comment (riyenas0925) [inline db/initdb.d/fosslight_create.sql:2281]: Code 940번으로 변경 (현재 914)

**Comment by riyenas0925** [comment]
@soimkim
review reflected 😄

**Comment by riyenas0925** [comment]
@soimkim 
query for existing db users

```sql
INSERT INTO `T2_CODE` (`CD_NO`, `CD_NM`, `CD_EXP`, `SYS_CD_YN`) VALUE
	('930', 'External Service Setting', 'System Detail Setting Code', 'Y');

INSERT INTO `T2_CODE_DTL` (`CD_NO`, `CD_DTL_NO`, `CD_DTL_NM`, `CD_SUB_NO`, `CD_DTL_EXP`, `CD_ORDER`, `USE_YN`) VALUES
	('909', '940', 'External Service Flag', '', 'N', 5, 'Y'),
	('930', '100', 'Github Token', '', 'github token', 1, 'Y');
```

**Review by soimkim — APPROVED** [review]

**Comment by soimkim** [comment]
@riyenas0925 ,
Please share the query statement to directly put Git_TOKEN value in DB without using Configuration UI.
ex) Query statement to save '**abcd1234**' with git_token.

**Comment by riyenas0925** [comment]
@soimkim 
Currently, there is already a query that puts a Github Token with the value "". change this value to an encrypted abcd1234 string?

```
('930', '100', 'Github Token', '', 'github token', 1, 'Y'); <- insert github token empty string
```

**Comment by soimkim** [comment]
No. 
I asked to leave an UPDATE query, considering that there will be a change to a query without using the Configuration tab later.

File: src/main/java/oss/fosslight/service/impl/SystemConfigurationServiceImpl.java [hunk 4]
```diff
@@ -138,6 +147,34 @@
...
+
+		if(tokenDetailMap != null) {
+			tokenAuthList.stream().map(c -> {
+				switch (c.getCdDtlNo()) {
+					case CoConstDef.CD_DTL_GITHUB_TOKEN:
+						String _token = (String) tokenDetailMap.get(CoConstDef.CD_DTL_GITHUB_TOKEN);
+						if(!StringUtil.isEmpty(_token)) {
+							String _encToken = null;
+							try {
+								_encToken = CryptUtil.encryptAES256(_token, CoConstDef.ENCRYPT_DEFAULT_SALT_KEY);
```
Comment (riyenas0925) [inline src/main/java/oss/fosslight/service/impl/SystemConfigurationServiceImpl.java:162]: @soimkim github token encrypted

**Pull request merged**